doc: Improve article on HTTPS

lorenzleutgeb opened the pull request:
This is a follow up on #9896.
- Adjust header to match `_page_title`
- Add instructions on deletion of CSRs and setting permissions
- Simplify some path expressions and commands
- Consqeuently use `~` instead of `${HOME}`
- Precise formulation ('key' vs. 'public key')
- Fix wrong indentation of output of `openssl req`
- Use dash ('--') instead of minus ('-')

**Remark on permissions:**

It's not a problem to `chmod 0400` the private keys, because the
Docker daemon runs as root (can read the file anyway) and the Docker
client runs as user.

lorenzleutgeb committed:
doc: Improve article on HTTPS

 * Adjust header to match _page_title
 * Add instructions on deletion of CSRs and setting permissions
 * Simplify some path expressions and commands
 * Consqeuently use ~ instead of ${HOME}
 * Precise formulation ('key' vs. 'public key')
 * Fix wrong indentation of output of `openssl req`
 * Use dash ('--') instead of minus ('-')

Remark on permissions:

It's not a problem to `chmod 0400` the private keys, because the
Docker daemon runs as root (can read the file anyway) and the Docker
client runs as user.

Signed-off-by: Lorenz Leutgeb <lorenz.leutgeb@gmail.com>

SvenDowideit commented on docs/sources/articles/https.md:
```diff
@@ -37,20 +37,20 @@ First generate CA private and public keys:
+    Organization Name (eg, company) [Internet Widgits Pty Ltd]:Docker Inc
+    Organizational Unit Name (eg, section) []:Boot2Docker
+    Common Name (e.g. server FQDN or YOUR name) []:your.host.com
+    Email Address []:Sven@home.org.au
```
This extra space was because the program output was indented - has that changed?

flowlo replied:
I got non-indented output with `OpenSSL 1.0.1j-fips 15 Oct 2014` on Fedora 21 and on an up-to-date MacBook (don't have access to it right now, so I can't give you the specific version).

fredlf replied:
It renders correctly (indented) on the site with the spaces in there. I think they're needed for our markdown renderer.

SvenDowideit replied:
@flowlo ah, sweet :)

flowlo replied:
Now you got me confused on this one. Did I break rendering by removing the spaces or is everything good?

fredlf replied:
Well, now I'm confused too! :-) 

I'll do some testing and try and figure it out.

fredlf replied:
Okay, the space is fine. You can leave it in. Sorry for the churn!

flowlo replied:
So you want me to revert my changes and put the spaces back in? :P

fredlf replied:
@flowlo No, it's fine as is.

SvenDowideit commented:
LGTM - @fredlf @jamtur01

jamtur01 commented:
LGTM

fredlf commented on docs/sources/articles/https.md:
```diff
@@ -96,6 +96,24 @@ Now sign the key:
+
+    $ rm -v client.csr server.csr
+
+With a default `umask` of 022 your secret keys will be *world-readable* and
```
Need a comma after "022", please.

fredlf commented on docs/sources/articles/https.md:
```diff
@@ -96,6 +96,24 @@ Now sign the key:
+writable for you and your group.
+
+To remove write permissions for your keys in order to protect them from accidental
+damage and make them only readable to you issue the following file mode changes:
```
Need to break up this run-on. Suggest something like "In order to protect your keys from accidental damage, you will want to remove their write permissions. To make them read-only, change file modes as follows:

flowlo replied:
@fredlf I know it's a bit late to say that as I already pushed, but I feel that the "only readable to you" (or maybe "readable to only you"?) part is important and semantically different to "read-only". Suggestions?

fredlf replied:
Oh, yeah, I agree, that is semantically different. Let's go with "...make
them only readable by you"

On Thu, Jan 8, 2015 at 2:55 PM, Lorenz Leutgeb <notifications@github.com>
wrote:

> In docs/sources/articles/https.md
> <https://github.com/docker/docker/pull/9952#discussion-diff-22689738>:
>
> > @@ -96,6 +96,24 @@ Now sign the key:
> >      Getting CA Private Key
> >      Enter pass phrase for ca-key.pem:
> >
> > +After generating `cert.pem` and `server-cert.pem` you can safely remove the
> > +two certificate signing requests:
> > +
> > +    $ rm -v client.csr server.csr
> > +
> > +With a default `umask` of 022 your secret keys will be *world-readable* and
> > +writable for you and your group.
> > +
> > +To remove write permissions for your keys in order to protect them from accidental
> > +damage and make them only readable to you issue the following file mode changes:
>
> @fredlf <https://github.com/fredlf> I know it's a bit late to say that as
> I already pushed, but I feel that the "only readable to you" (or maybe
> "readable to only you"?) part is important and semantically different to
> "read-only". Suggestions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/9952/files#r22689738>.
>

flowlo replied:
ACK. Landed in f9e2e47

fredlf commented:
Just a couple of editorial changes needed. Will LGTM and merge once those are done. Many thanks for the contribution!

lorenzleutgeb committed:
doc: Editorial changes as suggested by @fredlf

Refer to:
 * https://github.com/docker/docker/pull/9952#discussion_r22686652
 * https://github.com/docker/docker/pull/9952#discussion_r22686804

Signed-off-by: Lorenz Leutgeb <lorenz.leutgeb@gmail.com>

flowlo commented:
Okay, I amended changes as suggested by @fredlf

Could someone please take the time and check whether my statement

> It's not a problem to chmod 0400 the private keys, because the
> Docker daemon runs as root (can read the file anyway) and the Docker
> client runs as user.

holds by trying this out live?

Don't get me wrong: Surely I tried it and everything worked out on my end, but I'd like to share responsibility with you and make sure that I did not misdo anything in the process.

And as for comments, you'll have to bear with such long sentences. It's my mother tongue German that shines through :)

lorenzleutgeb committed:
doc: Minor semantical/editorial fixes in HTTPS article

"read-only" vs. "only readable by you"

Refer to:
https://github.com/docker/docker/pull/9952#discussion_r22690266

Signed-off-by: Lorenz Leutgeb <lorenz.leutgeb@gmail.com>

SvenDowideit commented:
yup, I'll test it soon - mmm, maybe next week

fredlf commented:
Ping @SvenDowideit Have you had a chance to validate this?

SvenDowideit commented:
sorry, not yet - 24hrs more :/

jamtur01 commented:
Ping @SvenDowideit

flowlo commented:
![Tumbleweed](http://img4.wikia.nocookie.net/__cb20140411060510/jackinthebox/images/2/2f/Tumbleweed.gif)

jamtur01 commented:
Sorry @flowlo the docs team is swamped with the 1.5 release. I am sure @SvenDowideit will get to this ASAP. 

Also I note this can't be automatically merged - I think you need to rebase.

fredlf commented:
Sorry, @flowlo @SvenDowideit is on PTO this week. And yes, we are swamped. Hence the lack of progress.

flowlo commented:
@fredlf @jamtur01 Sorry I'm not following this repo very closely, so I did not notice the existing load. Just wanted to make sure this does not peter out, no worries!

I just rebased and will add a small fix to the rebased version in the next minutes.

lorenzleutgeb committed:
Fix inconsistent formatting

Colon was bold, but regular at other occurences.

Blame cf27b310c4fc8d2c13ba181398a628d03e1e3c58

Signed-off-by: Lorenz Leutgeb <lorenz.leutgeb@gmail.com>

SvenDowideit commented on docs/sources/articles/https.md:
```diff
@@ -130,16 +148,13 @@ need to provide your client keys, certificates and trusted CA:
-    $ export DOCKER_HOST=tcp://:2376
-    $ export DOCKER_TLS_VERIFY=1
+    $ mkdir -pv ~/.docker
+    $ cp -v {ca,cert,key}.pem ~/.docker
```
I get `cp: cannot stat `{ca,cert,key}.pem': No such file or directory`

are you making a shell assumption here? (I'm using a busybox sh)

SvenDowideit commented:
ok, tested, and works for me on bash and fish, I won't worry about the sh test fail for now :)
